src/libraries/System.Private.CoreLib/src/System/Text/Encoding.cs

```diff
@@ -1140,7 +1140,7 @@ public static Stream CreateTranscodingStream(Stream innerStream, Encoding innerS
             // Special message to include fallback type in case fallback's GetMaxCharCount is broken
             // This happens if user has implemented an encoder fallback with a broken GetMaxCharCount
             throw new ArgumentException(
-                SR.Format(SR.Argument_EncodingConversionOverflowBytes, EncodingName, EncoderFallback.GetType()), "bytes");
+                SR.Format(SR.Argument_EncodingConversionOverflowBytes, GetType(), _codePage, EncoderFallback.GetType()), "bytes");
```

Comment:
Is the concrete type returned from GetType() here going to be meaningful to consumers?

Reply:
As the type is abstract I believe so

Reply:
The type name will be very confusing and will not be helpful in most cases. Can we get the encoding name inside try-catch block to avoid throwing case? I am not expecting the encoding name to not be throw most of the time too.  If we need to go with this change I would suggest at least get rid of the type name.

Reply:
Made an update with GetType out. This exception is rarely thrown and as mentioned in the description EncodingName has a surprisingly large dependency. I tried to simplify this corner case not to impact every .net app.